plugin/pki/path_venafi_cert_enroll.go

```diff
@@ -184,24 +196,50 @@ func (b *backend) pathVenafiCertObtain(ctx context.Context, logicalRequest *logi
 	}
 
 	var certId string
-	if !role.NoStore && role.StoreBy == storeByHASHstring {
+	// we currently ignore workflow for signingCSR because we are not supporting "hash" mode for it. That applies to every
+	// "hash" related workflow.
+	// we don't validate ignoring the local storage because we are making the "hash" mode independent of the prevent-reissue feature
+	if !role.NoStore && role.StoreBy == storeByHASHstring && !signCSR {
 		certId = getCertIdHash(*reqData, cfg.Zone, b.Logger())
 	}
 
-	if !role.IgnoreLocalStorage && role.StorePrivateKey == true && (role.StoreBy == storeBySerialString || role.StoreBySerial == true) {
+	if !role.IgnoreLocalStorage && role.StorePrivateKey && role.StoreBy == storeBySerialString && !signCSR {
 		// if we don't receive a logic response, whenever is an error or the actual certificate found in storage
 		// means we need to issue a new one
 		logicalResp := preventReissue(b, ctx, logicalRequest, reqData, &connector, role, cfg.Zone)
 		if logicalResp != nil {
 			return logicalResp, nil
 		}
-	} else if !role.IgnoreLocalStorage && role.StorePrivateKey == true && role.StoreBy == storeByHASHstring {
+	} else if !role.IgnoreLocalStorage && role.StorePrivateKey && role.StoreBy == storeByHASHstring && !signCSR {
 		logicalResp := preventReissueLocal(b, ctx, logicalRequest, reqData, role, certId)
 		if logicalResp != nil {
 			return logicalResp, nil
 		}
 	}
 
+	// if user is using store by hash
```

Comment:
@luispresuelVenafi we are missing the code of returning the call to Leader node since the  write is needed